modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/dao/JDBCResourceDAO.java

```diff
@@ -376,4 +381,237 @@ public boolean updateResource(String applicationKey, String resourceType, String
+
+        DeploymentConfig deploymentConfig = new DeploymentConfig();
+
+        try {
+            databaseConnection = AppFactoryDBUtil.getConnection();
+            getDeploymentPreparedStatement = databaseConnection.prepareStatement(SQLConstants.GET_DEPLOYEMENET);
+
+            String deploymentName = applicationId + "-" + stage;
+            getDeploymentPreparedStatement.setString(1, deploymentName);
+
+            deployementResultSet = getDeploymentPreparedStatement.executeQuery();
+
+            int deployementId = 0;
+
+            while (deployementResultSet.next()) {
+                deployementId = deployementResultSet.getInt(SQLParameterConstants.COLUMN_NAME_DEPLOYMENT_ID);
+                deploymentConfig.setDeploymentName(
+                        deployementResultSet.getString(SQLParameterConstants.COLUMN_NAME_DEPLOYMENT_NAME));
+                deploymentConfig.setReplicas(deployementResultSet.getInt(SQLParameterConstants.COLUMN_NAME_REPLICAS));
+            }
+
+            List<Container> containers = getContainers(deployementId, databaseConnection);
+            deploymentConfig.setContainers(containers);
+
+        } catch (SQLException e) {
+            String message = "Error while getting deployement config for application id : " + applicationId
+                    + " in stage : " + stage;
+            throw new AppFactoryException(message, e);
+        } finally {
+            AppFactoryDBUtil.closeResultSet(deployementResultSet);
+            AppFactoryDBUtil.closePreparedStatement(getDeploymentPreparedStatement);
+            AppFactoryDBUtil.closeConnection(databaseConnection);
+        }
+
+        return deploymentConfig;
+    }
+
+    /**
+     * Add containers to deployment config
+     *
+     * @param containerList   list of containers
+     * @param deployementName name of the deployement
+     * @param connection      database connection
+     */
+    private void addContainer(List<Container> containerList, String deployementName, Connection connection) {
+        PreparedStatement addContainerPreparedStatement = null;
+        try {
+            addContainerPreparedStatement = connection.prepareStatement(SQLConstants.ADD_CONTAINERS);
+            for (Container container : containerList) {
+                addContainerPreparedStatement.setString(1, container.getBaseImageName());
+                addContainerPreparedStatement.setString(2, container.getBaseImageVersion());
+                addContainerPreparedStatement.setString(3, deployementName);
+                addContainerPreparedStatement.executeUpdate();
+
+                List<ServiceProxy> serviceProxyList = container.getServiceProxies();
+                addServiceProxy(serviceProxyList, container, deployementName, connection);
+            }
+        } catch (SQLException e) {
+            String message = "Error while adding the database for container to deployment : " + deployementName;
+            log.error(message, e);
```

Comment:
Why we are not throwing the exception here? If you are not throwing the error, how does the calling method know, whether this operation was successful or not?